data/src/test/java/com/onboarding/data/RoomMapperTest.kt

```diff
@@ -0,0 +1,104 @@
+package com.onboarding.data
+
+import com.onboarding.data.database.entity.RoomCharacter
+import com.onboarding.data.database.entity.RoomLocationLink
+import com.onboarding.data.mapper.transform
+import com.onboarding.data.mapper.transformToRoom
+import com.onboarding.domain.entity.CharacterResult
+import com.onboarding.domain.entity.LocationLink
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class RoomMapperTest {
+
+    @Test
+    fun `location link transform to room`() {
+        val locationTransformed = characterResultTest.transformToRoom().location
+        assertEquals(STRING_VALUE, locationTransformed.locationName)
+        assertEquals(STRING_VALUE, locationTransformed.locationUrl)
+    }
+
+    @Test
+    fun `character result to room`() {
+        val characterResultTransformed = characterResultTest.transformToRoom()
+        assertEquals(INT_VALUE, characterResultTransformed.id)
+        assertEquals(STRING_VALUE, characterResultTransformed.name)
+        assertEquals(STRING_VALUE, characterResultTransformed.status)
+        assertEquals(STRING_VALUE, characterResultTransformed.species)
+        assertEquals(STRING_VALUE, characterResultTransformed.type)
+        assertEquals(STRING_VALUE, characterResultTransformed.gender)
+        assertEquals(STRING_VALUE, characterResultTransformed.image)
+        assertEquals(STRING_VALUE, characterResultTransformed.episodes.first())
+        assertEquals(STRING_VALUE, characterResultTransformed.characterUrl)
+        assertEquals(STRING_VALUE, characterResultTransformed.created)
```

Comment:
tip: you can compare two data class objects, no need to compare each parameter

Reply:
replaced fields for data class